### PR TITLE
WIP: Components for article

### PR DESCRIPTION
**Pull request by thomasthesecond**
This is a WIP, but since the PR is going to be larger than it should, it can be reviewed as I finish up. There's some clean up that I'll be doing as well.

### CODE REVIEW

File: src/components/action_sheet/index.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+import $ from "jquery";
+import urlencode from "urlencode";
+import "./action_sheet.scss";
+
+class ActionSheetComponent extends Component {
+  initialize() {
+    let self = this;
+    self.isActionSheetMenuHidden = false;
+
+    self.$el.find(".js-action-sheet-item").each(function(index, el) {
```
Comment (jcreamer898) [inline src/components/action_sheet/index.js]: use arrow methods and you don't need `self = this`

```js
.each((index, el) => {
```

File: src/components/related_guidebook/index.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+import $ from "jquery";
+
+import "./related_guidebook.scss";
+
+class RelatedGuidebookComponent extends Component {
+  initialize() {
+    if (!$("html").hasClass("ie9")) {
+      this._clampText();
+
+      $(window).on("resize", this._reclamp.bind(this));
```
Comment (jcreamer898) [inline src/components/related_guidebook/index.js]: debounce please

File: src/components/article_body/index.js [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    });
+  }
+  /**
+   * Format the post date with moment.js and append it to the bottom of the article
+   */
+  formatAndPositionDate() {
+    let date = this.$el.data("post-date"),
+        formattedDate = moment(date).format("MMMM YYYY");
+
+    $(".article-body__content").last()
```
Comment (jcreamer898) [inline src/components/article_body/index.js]: try not to use global `$`

File: src/components/action_sheet/index.js [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+  _share() {
+    let width = 550,
+        height = 420,
+        winHeight = $(window).height(),
+        winWidth = $(window).width(),
+        left,
+        top;
+
+    $(".action-sheet-menu").find(".action-sheet-control").each((index, el) => {
```
Comment (jcreamer898) [inline src/components/action_sheet/index.js]:
```js
this.events = {
  "click .action-sheet-control": "actionSheetControlClicked"
};
//...
actionSheetControlClicked() {
  // the codes
}
```

File: src/components/action_sheet/index.js [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+      $(el).on("click", function() {
+        let title = $(this).parents(".article").find("meta[itemprop=\"headline\"]")[0].content,
```
Comment (jcreamer898) [inline src/components/action_sheet/index.js]: `$(this).closest(".article")`

File: src/components/action_sheet/index.js [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+
+        left = Math.round((winWidth / 2) - (width / 2));
+        top = 0;
+
+        if (winHeight > height) {
+          top = Math.round((winHeight / 2) - (height / 2));
+        }
+
+        let windowOptions = "toolbar=no,menubar=no,location=yes,resizable=no,scrollbars=yes",
+            windowSize = "width=" + width + ",height=" + height + ",left=" + left + ",top=" + top;
```
Comment (jcreamer898) [inline src/components/action_sheet/index.js]:
```
`width=${width},height=${height},left=${left}`...
```

File: src/components/action_sheet/index.js [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+        winHeight = $(window).height(),
+        winWidth = $(window).width(),
+        left,
+        top;
+
+    $(".action-sheet-menu").find(".action-sheet-control").each((index, el) => {
+      $(el).on("click", function() {
+        let title = $(this).parents(".article").find("meta[itemprop=\"headline\"]")[0].content,
+            url = window.location.href,
+            network = $(this).data("network");
```
Comment (jcreamer898) [inline src/components/action_sheet/index.js]: Use $el instead of this
```js
let $el = $(event.currentTarget);
```

File: sass/_mixins.scss [hunk 8]
```diff
@@ -458,3 +458,22 @@
   top: 0;
   width: $width;
 }
+
+@mixin pop-out {
```
Comment (jcreamer898) [inline sass/_mixins.scss]: Document

File: sass/typography/_links_mixins.scss [hunk 9]
```diff
@@ -57,3 +57,36 @@
 
   padding: $gutter;
 }
+
+@mixin underline($color: $color-blue, $weight: .1rem, $offset: .2rem) {
```
Comment (jcreamer898) [inline sass/typography/_links_mixins.scss]: Document

**Comment by thomasthesecond** [comment]
Got an OK from @jcreamer898.

**Pull request merged**